lib/src/main/scala/spinal/lib/bus/misc/BusSlaveFactory.scala

```diff
@@ -502,28 +506,56 @@ trait BusSlaveFactory extends Area{
 
   /**
     * Emit on that a transaction when a write happen at address by using data placed at bitOffset in the word
+    *
+    * @param checkByteEnable do not trigger flow if byte enable is all zero.  See [[https://github.com/SpinalHDL/SpinalHDL/issues/1265]]
+   *                        for the discussion about this behaviour.
     */
-  def driveFlow[T <: Data](that      : Flow[T],
-                           address   : BigInt,
-                           bitOffset : Int = 0): Unit = {
+  def driveFlow[T <: Data](that           : Flow[T],
+                           address        : BigInt,
+                           bitOffset      : Int = 0,
+                           checkByteEnable: Boolean = false): Unit = {
 
     val wordCount = (bitOffset + widthOf(that.payload) - 1 ) / busDataWidth + 1
+    val byteEnable = writeByteEnable()
+    def assignValidNext(valid: Bool): Unit = {
+      if (byteEnable != null && checkByteEnable) {
+        when(byteEnable =/= 0) { valid := True }
+      } else {
+        valid := True
+      }
+    }
 
     if (wordCount == 1){
       that.valid := False
-      onWrite(address){ that.valid := True }
+      onWrite(address){ assignValidNext(that.valid) }
       nonStopWrite(that.payload, bitOffset)
     }else{
 
       assert(bitOffset == 0, "BusSlaveFactory ERROR [driveFlow] : BitOffset must be equal to 0 if the payload of the Flow is bigger than the data bus width")
 
       val regValid = RegNext(False) init(False)
-      onWrite(address + ((wordCount - 1) * wordAddressInc)){ regValid := True }
+      onWrite(address + ((wordCount - 1) * wordAddressInc)){ assignValidNext(regValid) }
       driveMultiWord(that.payload, address)
       that.valid := regValid
     }
   }
 
+  /**
+   * Emit on that a transaction when a write happen at address, by using data placed at bitOffset in the word.
+   * Block the write transaction until the transaction succeeds (stream becomes ready).
+   */
+  def driveStream[T <: Data](that: Stream[T], address: BigInt, bitOffset: Int = 0): Unit = {
+    val wordCount = (bitOffset + widthOf(that.payload) - 1) / busDataWidth + 1
+    onWritePrimitive(SizeMapping(address, wordCount * wordAddressInc), haltSensitive = false, null) {
+      when(!that.ready) {
+        writeHalt()
+      }
+    }
+    val flow = Flow(that.payloadType())
+    driveFlow(flow, address, bitOffset, checkByteEnable = true)
+    that << flow.toStream
```

Comment:
Here the flow.toStream might not wait the ready signal.
[This code](https://github.com/SpinalHDL/SpinalHDL/blob/2879ece341aa3d09c297c38c13270fe2d29786b8/lib/src/main/scala/spinal/lib/Flow.scala#L70) demostrate that.